api/v1beta1/azureimage_validation.go

```diff
@@ -62,14 +65,45 @@ func validateSingleDetailsOnly(image *Image, fldPath *field.Path) field.ErrorLis
 
 	if image.SharedGallery != nil {
 		if imageDetailsFound {
-			allErrs = append(allErrs, field.Forbidden(fldPath.Child("SharedGallery"), "SharedGallery cannot be used as an image ID or Marketplace images has been specified"))
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("SharedGallery"), "SharedGallery cannot be used as an image ID, Marketplace or ComputeGallery images has been specified"))
+		} else {
+			imageDetailsFound = true
+		}
+	}
+
+	if image.ComputeGallery != nil {
+		if imageDetailsFound {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("ComputeGallery"), "ComputeGallery cannot be used as an image ID, Marketplace or SharedGallery images has been specified"))
 		} else {
 			imageDetailsFound = true
 		}
 	}
 
 	if !imageDetailsFound {
-		allErrs = append(allErrs, field.Required(fldPath, "You must supply a ID, Marketplace or SharedGallery image details"))
+		allErrs = append(allErrs, field.Required(fldPath, "You must supply a ID, Marketplace, SharedGallery or ComputeGallery image details"))
+	}
+
+	return allErrs
+}
+
+func validateComputeGalleryImage(image *Image, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if image.ComputeGallery.Gallery == "" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("Gallery"), "", "Gallery cannot be empty when specifying an AzureComputeGalleryImage"))
```

Comment:
I don't think we need this validation since it's already done from the OpenAPI validation?

Reply:
Possibly, yeah. Though the OpenAPI validations are currently not tested (#2287). So I can either remove those validations here or add tests for them.

Reply:
My vote would be to remove them for simplicity since it's essentially unreachable code. I prefer to opt for OpenAPI validation whenever possible for straightforward validation and keep webhook validation for more involved validation logic.

Thanks for opening the issue to add tests. We have to think a bit more about it since it does add complexity to the tests and requires the extra binaries as you pointed out in the issue.

Reply:
Sure, make sense. Removed then and marking it as resolved.